frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -588,9 +604,29 @@ pub mod pallet {
         /// However, caller have to have deposit amount.
         /// TODO: weight
         #[pallet::weight(T::WeightInfo::payout_stakers_alive_staked(T::MaxStakings::get()))]
-        pub fn register(origin: OriginFor<T>, contract_id: T::AccountId) -> DispatchResult {
-            // TODO: impls
-            Ok(())
+        pub fn register(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            let operator = ensure_signed(origin)?;
+            let mut ledger = Self::ledger(&operator).ok_or(Error::<T>::NotController)?;
+            ensure!(
+                !RegisteredDapps::<T>::contains_key(&operator),
+                Error::<T>::AlreadyRegisteredContract
+            );
+            ensure!(
+                Self::is_contract(&contract_id),
+                Error::<T>::AddressIsNotContract
+            );
+            ensure!(
+                ledger.total >= T::RegisterDeposit::get(),
+                Error::<T>::MissingDeposit
+            );
+
+            RegisteredDapps::<T>::insert(&operator, contract_id.clone());
+            Self::deposit_event(Event::<T>::NewContract(contract_id, ledger.total));
```

Comment:
Did you want to emit event with a deposit amount?

Also, a thought - should deposit value be emitted here? It's a constant and it will be the same every time.

Reply:
ensure!(ledger.total >= T::RegisterDeposit::get() ...
deposit from the developer can be more than minimum.

Reply:
Ok but my comment wasn't about that.

Developer doesn't have control over how much the deposit is since it's a configurable constants.
It might be more than minimum, sure. But it will always be the same, it's not like it's an argument for this method.
So it seems redundant to report it.

What about the first comment - currently you're emitting the total bonded amount.
Is that intentional?